packages/shared/src/models-hdf5.ts

```diff
@@ -74,7 +74,7 @@ export type NumArrayDataset = Dataset<ArrayShape, NumericType>;
 
 export type Shape = ArrayShape | ScalarShape | null;
 export type ArrayShape = number[];
-export type ScalarShape = never[];
+export type ScalarShape = [];
```

Comment:
I think the empty tuple type makes more sense here (instead of the empty array type) and is perhaps a little stricter. Doesn't make a difference in practice though, so sorry for the noise.

Ideally, `ArrayShape` should be `[number, ...number]` to enforce at least one number and to make sure `assertArrayShape` actually excludes the empty tuple (which it doesn't at the moment, I realised), but this led to some problems... Might investigate another time.